pennylane/templates/subroutines/commuting_evolution.py

```diff
@@ -130,50 +132,85 @@ def __init__(self, hamiltonian, time, frequencies=None, shifts=None, id=None):
             recipe = qml.math.stack([c, qml.math.ones_like(c), s]).T
             self.grad_recipe = (recipe,) + (None,) * len(hamiltonian.data)
             self.grad_method = "A"
-
-        self._hyperparameters = {
-            "hamiltonian": hamiltonian,
-            "frequencies": frequencies,
-            "shifts": shifts,
-        }
-
-        super().__init__(time, *hamiltonian.parameters, wires=hamiltonian.wires, id=id)
-
-    def queue(self, context=qml.QueuingManager):
-        context.remove(self.hyperparameters["hamiltonian"])
-        context.append(self)
-        return self
+        else:
+            self.grad_recipe = (None,) * (1 + len(hamiltonian.data))
+
+        super().__init__(hamiltonian, scalar=time, id=id)
+        self.hyperparameters["frequencies"] = frequencies
+        self.hyperparameters["shifts"] = shifts
+
+    @property
+    def parameter_frequencies(self):
+        """Compute the parameter frequencies of CommutingEvolution."""
+        # TODO: Fix the following by exploiting the structure of CommutingEvolution
+        # Note that because of the coefficients of the Hamiltonian, we do not have
+        # parameter_frequencies even if "frequencies" are provided at initialization!
+        raise ParameterFrequenciesUndefinedError(
+            "CommutingEvolution has no parameter frequencies defined."
+        )
+
+    @property
+    def _queue_category(self):
+        return "_ops"
+
+    # pylint: disable=arguments-renamed, invalid-overridden-method
+    @property
+    def has_matrix(self):
+        return False
 
     @staticmethod
-    def compute_decomposition(
-        time, *_, wires, hamiltonian, **__
-    ):  # pylint: disable=arguments-differ,unused-argument
-        r"""Representation of the operator as a product of other operators.
+    def _matrix(scalar, mat):
+        return qml.math.expm(-1j * scalar * mat)
 
-        .. math:: O = O_1 O_2 \dots O_n.
+    def matrix(self, wire_order=None) -> TensorLike:
+        """Raise a MatrixUndefinedError for now to force decomposition on DefaultQubit."""

```

Comment:
Potentially instead we can treat this like `GroverOperator` and `QFT` and manually say "decompose CommutingEvolution".